site/src/contexts/ProxyContext.tsx

```diff
@@ -6,25 +6,60 @@ import {
+  useCallback,
   useContext,
+  useEffect,
   useState,
 } from "react"
 import { ProxyLatencyReport, useProxyLatency } from "./useProxyLatency"
 
-interface ProxyContextValue {
+export interface ProxyContextValue {
+  // proxy is **always** the workspace proxy that should be used.
+  // The 'proxy.selectedProxy' field is the proxy being used and comes from either:
+  //   1. The user manually selected this proxy. (saved to local storage)
+  //   2. The default proxy auto selected because:
+  //    a. The user has not selected a proxy.
+  //    b. The user's selected proxy is not in the list of proxies.
+  //    c. The user's selected proxy is not healthy.
+  //   3. undefined if there are no proxies.
+  //
+  // The values 'proxy.preferredPathAppURL' and 'proxy.preferredWildcardHostname' can
+  // always be used even if 'proxy.selectedProxy' is undefined. These values are sourced from
+  // the 'selectedProxy', but default to relative paths if the 'selectedProxy' is undefined.
   proxy: PreferredProxy
+
+  // userProxy is always the proxy the user has selected. This value comes from local storage.
+  // The value `proxy` should always be used instead of `userProxy`. `userProxy` is only exposed
+  // so the caller can determine if the proxy being used is the user's selected proxy, or if it
+  // was auto selected based on some other criteria.
+  //
+  // if(proxy.selectedProxy.id === userProxy.id) { /* user selected proxy */ }
+  // else { /* proxy was auto selected */ }
+  userProxy?: Region
+
+  // proxies is the list of proxies returned by coderd. This is fetched async.
+  // isFetched, isLoading, and error are used to track the state of the async call.
   proxies?: Region[]
-  proxyLatencies?: Record<string, ProxyLatencyReport>
-  // isfetched is true when the proxy api call is complete.
+  // isFetched is true when the 'proxies' api call is complete.
   isFetched: boolean
-  // isLoading is true if the proxy is in the process of being fetched.
   isLoading: boolean
   error?: Error | unknown
+  // proxyLatencies is a map of proxy id to latency report. If the proxyLatencies[proxy.id] is undefined
+  // then the latency has not been fetched yet. Calculations happen async for each proxy in the list.
+  // Refer to the returned report for a given proxy for more information.
+  proxyLatencies: Record<string, ProxyLatencyReport>
+  // setProxy is a function that sets the user's selected proxy. This function should
+  // only be called if the user is manually selecting a proxy. This value is stored in local
+  // storage and will persist across reloads and tabs.
   setProxy: (selectedProxy: Region) => void
+  // clearProxy is a function that clears the user's selected proxy.
+  // If no proxy is selected, then the default proxy will be used.
+  clearProxy: () => void
 }
 
 interface PreferredProxy {
-  // selectedProxy is the proxy the user has selected.
+  // selectedProxy is the preferred proxy being used. It is provided for
```

Comment:
Let's call this interface `Proxy`. `PreferredProxy` makes it sound like it's the `UserProxy` (it's what the user preferred).

Reply:
Sure!